### ISSUE TITLE
Don't allow deleting inbound shipment lines for transfers

### ISSUE THREAD

**Issue by marthakeezy** [msupply-foundation/open-msupply#5120]
## What went wrong? 😲

Screenshot -> [here](https://drive.google.com/drive/folders/1BN7odKHMWxWhVex7WsnvikyREu2xC445)

Database uploaded [here](https://mega.nz/fm/NY5BWSjS)

## Expected behaviour 🤔

Error should be user-friendly :)

## How to Reproduce 🔨

Steps to reproduce the behaviour:

1. Create an Outbount Shipment from another store, mark it as Shipped so it will appear as Inbound Shipment in the testing store
2. Confirm the item as Delivered
3. Select the item -> click on Actions -> try to remove it from the shipment

## Your environment 🌱

- Open mSupply Version: V2.3.0-RC8
- Legacy mSupply Central Server Version: V7.17.10

- Platform: Tablet (apk)

- Database type: SQLite3

**Comment by lache-melvin** [comment]
Kia ora @marthakeezy - could you share your DB again here for us to investigate? I can't replicate the issue, but we definitely shouldn't be having this kind of error on deleting shipment lines. Wondering if this is somehow related to the shipment-not-transferring issue you had a few days ago...

Setting as high priority to investigate the cause!

**Comment by marthakeezy** [comment]
Can be found [here](https://mega.nz/fm/Nc4nAIKa) :)

**Comment by andreievg** [comment]
deletion shouldn't be allowed on any transfer lines. so even though error seems quite bad, the use case is rare (shouldnt' be possible). proposed fix is to disallow deletion of invoice lines, also need to validate this

**Comment by aimee-mcneil-melville** [comment]
Consensus on this: Shouldn't be able to delete lines on a transferred inbound shipment

**Comment by lache-melvin** [comment]
Yep per Aimee's message, no deletes of any lines on a transferred inbound shipment. We will continue to allow adding items to transferred shipments, as there are some use cases where this is needed. Bit odd to allow adds but not deletes, but ok for now 🙏 

- [ ] Disable the delete selected lines action if is a transferred inbound shipment. Check out other areas of inbound shipment, like where statuses are handled - we derive manual or transferred based on `linked_shipment`/`linked_invoice`
- [ ] Add extra check in service layer (stock_in_line delete) that invoice does not have a linked invoice - be sure to add a test!

**Comment by marthakeezy** [comment]
# 🧪 Testing



`V2.5.0-RC6`

- [ ] Create an Outbound Shipment from a store to another of your stores, mark it as Shipped so it will appear as Inbound Shipment in the testing store
- [ ] Go to the store that the shipment was made to -> Inbound Shipments
- [ ] Confirm the shipment as Delivered
- [ ] Select an item -> click on Actions -> `Delete` should be disabled
